stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/repository/RepositoryFacade.kt

```diff
@@ -155,4 +157,49 @@ internal class RepositoryFacade constructor(
             }
         }
     }
+
+    internal companion object {
+        private var instance: RepositoryFacade? = null
+
+        internal fun initialise(factory: RepositoryFactory, scope: CoroutineScope, defaultConfig: Config) {
+            instance = create(factory, scope, defaultConfig)
+        }
+
+        /**
+         * Use it get create and refresh the singleton of [RepositoryFacade]. This is needed when we currentUser of the
+         * SDK is changes.
+         *
+         * Todo: Improve documentation
+         */
+        internal fun refreshAndGet(factory: RepositoryFactory, scope: CoroutineScope, defaultConfig: Config) =
+            create(factory, scope, defaultConfig).also { facade -> instance = facade }
+
+        internal fun get(): RepositoryFacade = instance ?: throw IllegalStateException(
+            "RepositoryFacade must be initialised first"
+        )
+
+        @VisibleForTesting
+        internal fun create(factory: RepositoryFactory, scope: CoroutineScope, defaultConfig: Config): RepositoryFacade {
```

Comment:
It is nice for unit test to be able to create an instance of it, but not as a singleton. Otherwise, the state is shared between unit tests which creates a mess.